app/controllers/PaymentScheduleController.scala

```diff
@@ -65,9 +66,13 @@ class PaymentScheduleController @Inject() (
         logErrorAndRouteToDefaultPageF(j)
 
       case j: Journey.AfterSelectedPaymentPlan =>
-        journeyService.updateHasCheckedPaymentPlan(j.journeyId)
-          .map(_ => Redirect(routes.BankDetailsController.typeOfAccount))
-
+        for {
+          _ <- journeyService.updateHasCheckedPaymentPlan(j.journeyId)
+          _ = j match {
+            case j1: Journey.Stages.ChosenPaymentPlan => auditService.auditPaymentPlanBeforeSubmission(j1)
+            case _                                    => JourneyLogger.info("Nothing to audit, so I won't")
```

Comment:
probably don't need this log at info level - maybe debug with a mention of what stage it is at so we've got the info if we need it?

Reply:
Will do